File: backend/api/src/database/chat-history/domain/chat-history.ts
import {MessageSender} from "@prisma/client";

export class ChatHistory {
    id: string;
    userId: string;
    message: string;
    sender: MessageSender;
    isQuestionResponse: boolean;
    isUserInitiated: boolean;
    analyzedByLlm: boolean;
    timestamp: Date;
}

File: frontend/app/lib/data/network/model/chat_history.dart
class ChatHistory {
  final String id;
  final String userId;
  final String roomId;
  final String message;
  final String sender; // 'USER' or 'AI'
  final DateTime timestamp;

  ChatHistory({
    required this.id,
    required this.userId,
    required this.roomId,
    required this.message,
    required this.sender,
    required this.timestamp,
  });

  factory ChatHistory.fromJson(Map<String, dynamic> json) => ChatHistory(
    id: json['id'],
    userId: json['userId'],
    roomId: json['roomId'],
    message: json['message'],
    sender: json['sender'],
    timestamp: DateTime.parse(json['timestamp']),
  );
}